recipe/build.sh

```diff
@@ -8,9 +8,20 @@ cmake \
 	-DBUILD_PYTHON_BINDINGS=ON \
 	-DBUILD_CLI_EXECUTABLES=ON \
 	-DBUILD_SHARED_LIBRARIES=ON \
-	-DFORCE_CXX11=OFF \
 	-DBUILD_TESTS=OFF \
-	../
+	-DMATLAB_BINDINGS=OFF \
+	-DDOWNLOAD_STB_IMAGE=OFF \
+	-DDOWNLOAD_ENSMALLEN=OFF \
+	-DPYTHON_EXECUTABLE=${PYTHON} \
+	-DUSE_OPENMP=ON \
+	-DBOOST_ROOT=$PREFIX \
+	-DENSMALLEN_INCLUDE_DIR=$PREFIX/include \
+	-DARMADILLO_INCLUDE_DIR=$PREFIX/include \
+	-DARMADILLO_LIBRARY=$PREFIX/lib \
+	-DBOOST_INCLUDEDIR=$PREFIX/include \
+	-DBoost_NO_SYSTEM_PATHS=ON \
+	-DBOOST_LIBRARYDIR=$PREFIX/lib \
```

Comment:
You might consider using `-DBoost_NO_BOOST_CMAKE=1`---I found that I had lots of problems with Boost 1.70.0's CMake scripts.  I hadn't debugged it any further, but I thought maybe posting that here might be a help. :+1:

Reply:
Thank you very much! I am going to test it

Reply:
I think I will try to pin ``boost <1.70 `` to see if I can build it

Reply:
Even with boost 1.69 is not working

Reply:
Try specifying *only* `-DBoost_NO_BOOST_CMAKE` and maybe, if needed, `-DBOOST_ROOT` to specify the root of the Boost installation if CMake can't find it.  The build logs indicate that the directory `$PREFIX/include` is used as the Boost include directory, which doesn't seem right---shouldn't `$PREFIX` be expanded to something?

Anyway, I don't know conda, so maybe my input isn't helpful, I'm not sure.

Reply:
But the things is, the ``FindBoost`` is finding ``boost``, the problem what I am thinking that is happening (which I saw in other builds as well) is that ``cmake`` is not including the headers and because of that the compiler is not being able to find it. ``cmake`` is pretty bad to work with sysroot/virtual envs, they are trying to improve it, but it still sucks :(

Reply:
When I looked at the output of one of the failed builds, it seemed like `$PREFIX` wasn't being expanded in the CMake call.  Specifically I would expect a line like this:

```
-- Found PythonInterp: /usr/bin/python (found version "2.7.16")
```

but in that output I was seeing, e.g.,

```
-- Found PythonInterp: $PREFIX/python
```

which to me is a sign that `$PREFIX` didn't get expanded correctly.

However, remember, I'm not familiar with the system, so I'm not certain if that is the right way to debug.  You can always build with `VERBOSE=1 make` to see *exactly* what CMake is assembling as a compiler call, and that could be informative.

Reply:
> When I looked at the output of one of the failed builds, it seemed like `$PREFIX` wasn't being expanded in the CMake call. Specifically I would expect a line like this:
> 
> ```
> -- Found PythonInterp: /usr/bin/python (found version "2.7.16")
> ```
> 
> but in that output I was seeing, e.g.,
> 
> ```
> -- Found PythonInterp: $PREFIX/python
> ```
> 
> which to me is a sign that `$PREFIX` didn't get expanded correctly.
> 
> However, remember, I'm not familiar with the system, so I'm not certain if that is the right way to debug. You can always build with `VERBOSE=1 make` to see _exactly_ what CMake is assembling as a compiler call, and that could be informative.

That is fine. For some reason, it is not finding the boost headers.

Reply:
The boost is not being included 
```
$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++;-Dmlpack_EXPORTS;-DARMA_NO_DEBUG;-DNDEBUG;-DBOOST_TEST_DYN_LINK;-DHAS_OPENMP;-DMLPACK_GIT_VERSION;-fvisibility-inlines-hidden;-std=c++17;-fmessage-length=0;-march=nocona;-mtune=haswell;-ftree-vectorize;-fPIC;-fstack-protector-strong;-fno-plt;-O2;-ffunction-sections;-pipe;-fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/mlpack-3.2.1;-fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix;-Wall;-Wextra;-ftemplate-depth=1000;-Wno-unused-function;-O3;-fopenmp;-O3;-std=gnu++11;-fPIC;-I$SRC_DIR/src;-I$SRC_DIR/build/src/mlpack/..;-x;c++-header;-c;$SRC_DIR/build/src/mlpack/cotire/mlpack_CXX_prefix.hxx;-o;$SRC_DIR/build/src/mlpack/cotire/mlpack_CXX_prefix.hxx.gch
```

Reply:
The include directories I see there are only `-I$SRC_DIR/src;-I$SRC_DIR/build/src/mlpack/..`.  When I look at the output there are lots of strange warnings.  Here's a really weird one:

```
WARNING: Target "mlpack" requests linking to directory "/home/conda/feedstock_root/build_artifacts/mlpack_1570035530398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib".  Targets may link only to libraries.  CMake is dropping the item.
```

It's really confusing to me how that might have happened.  It seems like somehow there is actually a directory called `$PREFIX`?  (i.e., `$PREFIX` is not an expanded environment variable but instead an actual directory with the `$` character in it?)  To me this is really confusing and it seems like that would probably be the source of the problems.  If you can get a copy of the `CMakeCache.txt` that is getting generated that could help also because you could see the directory that is being chosen as `BOOST_INCLUDE_DIR` and see if that actually includes the file `boost/serialization/serialization.hpp`.

Reply:
That warning is strange, indeed.
But that part related to ``$PREFIX`` seems to be right
https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html